### ISSUE TITLE
Bug Report: Variable defined as covariate in `tbl_uvregression` is reported in table output

### ISSUE THREAD

**Issue by Eduardo-Auer** [ddsjoberg/gtsummary#1886]
Dear Mr. Sjoberg,
Dear gtsummary Team,

Hello,
I am using package gtsummary version 2.0.0 under R version 4.4.1 (Windows) and I noticed that when the variable, previously defined as a covariate in the formula argument of tbl_uvregression function, was reported in the table output. I also remembered it did not happen in previous versions and only noticed this issue after an update. 

``` r
#load packages
    
    library(tidyverse)
    library(gtsummary)
    
    #load data
    df <- read.table("https://archive.ics.uci.edu/ml/machine-learning-databases/heart-disease/processed.hungarian.data",sep = ",", header=TRUE)
    
    colnames(df)<-c("var_1",
                    "var_2",
                    "var_3",
                    "var_4",
                    "var_5",
                    "var_6",
                    "var_7",
                    "var_8",
                    "var_9",
                    "var_10",
                    "var_11",
                    "var_12",
                    "var_13",
                    "var_14")
    
    #univariate regression
    
    df <- df %>% mutate_at(vars(var_1,
                                var_3,
                                var_4,
                                var_5,
                                var_8,
                                var_10), as.numeric)
#> Warning: There were 3 warnings in `mutate()`.
#> The first warning was:
#> ℹ In argument: `var_4 = .Primitive("as.double")(var_4)`.
#> Caused by warning:
#> ! NAs introduzidos por coerção
#> ℹ Run `dplyr::last_dplyr_warnings()` to see the 2 remaining warnings.
    
    df %>%
        select(var_1,
               var_3,
               var_4,
               var_5,
               var_8,
               var_10) %>% 
        tbl_uvregression(
            method = lm,
            y = var_1,
            exponentiate = FALSE,
            hide_n = FALSE,
            formula = "{y} ~ {x} + var_3",
            pvalue_fun = function(x) {
                if_else(
                    is.na(x), 
                    NA_character_,
                    if_else(x < 0.001, format(x, digits = 2, scientific = TRUE), format(round(x, 3), scientific = F))
                )
            }) %>%
        add_q(method = "fdr") %>% 
        bold_p(q = TRUE, t = 0.05) %>% 
        sort_p(q = TRUE) %>% 
        modify_header(label ~ "Variables", p.value ~ "p", estimate ~ "Beta", conf.low ~ "CI 95%", q.value = "pcorr") %>% 
        as_flex_table()
```

<img src="https://i.imgur.com/uw2MeE9.png" width="953" />

<sup>Created on 2024-08-03 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by ddsjoberg** [comment]
Thank you @Eduardo-Auer for reporting this! It should be fixed in PR https://github.com/ddsjoberg/gtsummary/pull/1887. Can you take a look and confirm, please?

**Comment by Eduardo-Auer** [comment]
> Thank you @Eduardo-Auer for reporting this! It should be fixed in PR #1887. Can you take a look and confirm, please?

Thank you so much for the quick response! Yes, I will wait for the fix in the next update.
